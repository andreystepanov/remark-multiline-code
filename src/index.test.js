import multilineCode from './'
import remark from 'remark'

const parse = data =>
  remark()
    .use(multilineCode)
    .use({ settings: { fences: true } })
    .processSync(data)
    .toString()

test('defined', () => {
  expect(multilineCode).toBeDefined()
  expect(typeof multilineCode).toBe('function')
})

test("ignores is there's more than 1 child in paragraph", () => {
  const md = parse('Some text with `code`')
  expect(md).toMatchSnapshot()
})

test('Single inlineCode inside blockquote', () => {
  const md = parse('> `code`')
  expect(md).toMatchSnapshot()
})

test('turnes inlineCode into code', () => {
  const md = parse('Some text with:\n\n`code`')
  expect(md).toMatchSnapshot()
})
