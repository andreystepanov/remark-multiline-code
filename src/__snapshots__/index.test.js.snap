// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ignores if there's some other content in the paragraph 1`] = `
"\`code\` and some text
"
`;

exports[`ignores inlineCode inside blockquote 1`] = `
"> \`code\`

> Quote
>
> > Inner quote
> > \`code\`
"
`;

exports[`ignores is there's more than 1 child in paragraph 1`] = `
"Some text with \`code\`
"
`;

exports[`turnes inlineCode into code 1`] = `
"Some text with:

\`\`\`
code
\`\`\`
"
`;
